### PR TITLE
Access services in specific cluster

### LINKED ISSUE #200: Access remote services in a specific cluster explicitly
I'm looking at options for deploying submariner and lighthouse in pre-existing clusters.  If two clusters `east` and `west` have the same (default) dns suffix `cluster.local` is there a way for pods in cluster `east` to access services in cluster `west` explicitly using some form of a forward plugin config for CoreDNS in `east` so that lookups for `*.svc.west.local` (or something else) are sent to lighthouse and will only match services exported by the `west` cluster?

Note that this is a little different from but not necessarily incompatible with the current (0.4.0) design of Lighthouse which supports multi-cluster services which can be exported from (and therefore serviced by) one or more clusters.  Such multi-cluster services are visible in all clusters, and discoverable via a new `supercluster.local` domain name, and it's assumed that namespaces that export such services will be globally unique across the set of clusters.  (I assume this looks like `_service_._namespace_.svc.supercluster.local` ?)

However, if the clusters are pre-existing, it's possible that they may have existing namespaces that are not globally unique, such as `kube-system` or `monitoring` (or in my case `kafka`).  So I'm trying to connect to a specific remote service in a specific remote cluster, not a "multi-cluster" service that may exist in one or more clusters.  This makes the use case more similar to the pre-0.4.0 design of Lighthouse, but since both clusters are pre-existing, they do not have unique DNS names and both consider themselves to be `.cluster.local`.

### PR DESCRIPTION
**Pull request by vthapar**
Fixes #200

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

File: pkg/serviceimport/map.go [hunk 1]
```diff
@@ -70,6 +70,11 @@ func (m *Map) GetIPs(namespace, name string, checkCluster func(string) bool) ([]
 		return nil, false
 	}
 
+	if cluster != "" {
+		ips, found := clusterIPs[cluster]
+		return ips, found
```
Comment (tpantelis) [inline pkg/serviceimport/map.go:75]:
```suggestion
		return clusterIPs[cluster]
```

File: plugin/lighthouse/handler_test.go [hunk 2]
```diff
@@ -407,15 +420,29 @@ func testHeadlessService() {
...
-		It("should succeed and write all IPs as A records in response", func() {
-			executeTestCase(lh, rec, test.Case{
-				Qname: service1 + "." + namespace1 + ".svc.cluster.local.",
-				Qtype: dns.TypeA,
-				Rcode: dns.RcodeSuccess,
-				Answer: []dns.RR{
-					test.A(service1 + "." + namespace1 + ".svc.cluster.local.    0    IN    A    " + serviceIP),
-					test.A(service1 + "." + namespace1 + ".svc.cluster.local.    0    IN    A    " + serviceIP2),
-				},
+		When("query for headless service", func() {
```
Comment (tpantelis) [inline plugin/lighthouse/handler_test.go:423]:
```suggestion
		When("no cluster is requested", func() {
```

File: plugin/lighthouse/handler_test.go [hunk 3]
```diff
@@ -407,15 +420,29 @@ func testHeadlessService() {
...
+					Qtype: dns.TypeA,
+					Rcode: dns.RcodeSuccess,
+					Answer: []dns.RR{
+						test.A(service1 + "." + namespace1 + ".svc.clusterset.local.    0    IN    A    " + serviceIP),
+						test.A(service1 + "." + namespace1 + ".svc.clusterset.local.    0    IN    A    " + serviceIP2),
+					},
+				})
+			})
+		})
+		When("query for headless service in specific cluster", func() {
```
Comment (tpantelis) [inline plugin/lighthouse/handler_test.go:436]:
```suggestion
		When("requested for a specific cluster", func() {
```

File: plugin/lighthouse/handler_test.go [hunk 4]
```diff
@@ -407,15 +420,29 @@ func testHeadlessService() {
...
+			It("should succeed and write one IP as A record in response", func() {
```
Comment (tpantelis) [inline plugin/lighthouse/handler_test.go:437]:
```suggestion
			It("should succeed and write the cluster's IP as A record in response", func() {
```

File: pkg/serviceimport/map_test.go [hunk 5]
```diff
@@ -163,7 +163,7 @@ var _ = Describe("ServiceImport Map", func() {
```
Comment (tpantelis) [inline pkg/serviceimport/map_test.go:163]: I think we should add a test case for non-empty cluster to be complete.

File: pkg/serviceimport/map.go (hunk 1 above)
Comment (vthapar) [reply pkg/serviceimport/map.go:75]: This gives `Not enough arguments to return`

**Review by tpantelis — APPROVED** [review]

File: pkg/serviceimport/map_test.go (hunk 5 above)
Comment (vthapar) [reply pkg/serviceimport/map_test.go:191]: Done

**Review by tpantelis — DISMISSED** [review]

**Review by aswinsuryan — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]
